gnu-apl-spreadsheet.el

```diff
@@ -123,6 +129,9 @@ value of VARNAME to the content of the spreadsheet."
 values as the spreadsheet in the current buffer."
   (concat (gnu-apl-make-array-loading-instructions varname)))
 
+(defvar ses--numrows)                   ;ses.el
```

Comment:
I'm not at all a fan of defining variables with names from other packages. This should not be needed.

Reply:
Without these declarations, I seem to get the following issues:

```
emacs -Q --batch -L . -f batch-byte-compile gnu-apl-spreadsheet.el

In gnu-apl-make-array-loading-instructions:
gnu-apl-spreadsheet.el:139:17: Warning: reference to free variable
    ‘ses--numrows’
gnu-apl-spreadsheet.el:140:17: Warning: reference to free variable
    ‘ses--numcols’
gnu-apl-spreadsheet.el:152:42: Warning: reference to free variable
    ‘ses--cells’
```

Reply:
That's because these variables are buffer-local to SES buffers. They shouldn't be visible globally.

Reply:
But in this case they have to be "visible" as gnu-apl-spreadsheet.el isn't part of SES. After all, the `--` indicates that the variables are not actually part of the interface that SES provides. They have to at least be declare for the byte compiler, which doesn't mean they are bound, so I don't think this is a real issue.

Reply:
True. Having to access internal symbols is not great. An issue is that SES doesn't really have an external API. Also, I'm pretty sure that when SES is compiled, you're going to see the same warnings.

Doing a `defvar` on another package's symbol is still bad, and while extremely unlikely, could still cause problems outside of gnu-apl-mode.

If you really want to get rid of that warning, one can always work around it using `eval`. Horrible, I know, but better than messing with another package's symbols.

Reply:
I hesitate to use eval, because I don't think that the defvar approach is _that_ bad, considering that these are just declarations and not definitions. But if you insist, I'll make the change.

> Also, I'm pretty sure that when SES is compiled, you're going to see the same warnings.

It doesn't seem to raise any issues on my system (using a recent checkout of 28.0.60).

Reply:
Interesting. Because you had to add these declarations in order to get rid of these warnings, but SES doesn't have them, yet don't have the warnings. Why is it different? I'd need to look at what SES is doing.

Reply:
OK, I'm frustrated by this one. I will merge your changes now, and I will at some point try to find out what is going on so the declaration can be avoided